src/ctia/lib/riemann.clj

```diff
@@ -0,0 +1,167 @@
+(ns ctia.lib.riemann
+  (:require [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [riemann.client :as riemann]
+            [ctia.properties :as prop]
+            [ctia.lib.utils :as utils])
+  (:import [clojure.lang ExceptionInfo]))
+
+;; based on riemann-reporter.core/request-to-event
+(defn request->event
+  [request extra-fields]
+  (into {:uri (str (:uri request))
+         :_params (utils/safe-pprint-str (:params request))
+         :remote-addr (str (if-let [xff (get-in request [:headers "x-forwarded-for"])]
+                             (peek (str/split xff #"\s*,\s*"))
+                             (:remote-addr request)))
+         :request-method (str (:request-method request))
+         :identity (:identity request)
+         :jwt (:jwt request)}
+        extra-fields))
+
+(defn ms-elapsed
+  "provide how much ms were elapsed since `nano-start`."
+  [nano-start]
+  (/ (- (System/nanoTime) nano-start) 1000000.0))
+
+;; based on riemann-reporter.core/send-request-metrics
+(defn- send-request-logs [send-event-fn request extra-fields]
+  (let [event (request->event request extra-fields)]
+    (try
+      (send-event-fn event)
+      (catch Exception e
+        (log/warnf "A Problem occured while sending request metrics event:\n\n%s\n\nException:\n%s"
+                   (utils/safe-pprint-str event)
+                   (utils/safe-pprint-str e))
+        (when (nil? send-event-fn)
+          (log/warn "The send-event-fn looks empty. This is certainly due to a configuration problem or perhaps simply a code bug."))))))
+
+;; should align with riemann-reporter.core/extract-infos-from-event
+(defn extract-infos-from-event
+  [event]
+  (let [search-event get-in]
```

Comment:
So the `get-in` will not work here. Because I thin the `event` is already flattened when this function is called.

I think, if you are lucky, the function that might work is to rename all the keys starting with `identity-` to remove this `identity-` prefix. So replacing:

```clojure
{:identity-user-id ,,, 
 :identity-user-name ,,,
 :foo ,,,}

;; by

{:user-id ,,,
 :user-name ,,,
 :foo ,,,}
```

As I said, as a first step you could copy the `search-event` function verbatim from IROH, then improve build a better event from request before calling the `send-event` (see my other comments).